src/dnmd/deltas.c

```diff
@@ -1,125 +1,314 @@
 #include "internal.h"
 
-static bool merge_into_base(mdcxt_t* cxt, mdstream_t* base, mdstream_t* delta, size_t delta_offset)
+static bool append_heaps_from_delta(mdcxt_t* cxt, mdcxt_t* delta)
 {
-    assert(cxt != NULL);
-    assert(base != NULL);
-    assert(delta != NULL && delta->size > 0);
-
-    // Check if we are only copying a portion of the delta.
-    size_t delta_size = delta->size;
-    if (delta_offset != 0)
-    {
-        if (delta->size < delta_offset)
-            return false;
-        delta_size -= delta_offset;
-    }
+    if (delta == NULL)
+        return false;
```

Comment:
```suggestion
    assert(delta != NULL);
```

I don't think this needs a check.